environment/stats/src/cloud/benchflow/collectors/stats.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+	"sync"
+	"github.com/fsouza/go-dockerclient"
+	"github.com/minio/minio-go"
+)
+
+type Container struct {
+	ID           string
+	statsChannel chan *docker.Stats
+	doneChannel chan bool
+	stopChannel chan bool
```

Comment:
This situation should not be possible because you use the collecting variable to block any new start, if there is an already running collection. This variable is set to false only after all the routines have exited (using the Wait instruction).